tests/std/tests/P0645R10_text_formatting_args/test.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <concepts>
+#include <cstddef>
+#include <format>
+#include <memory>
+#include <string_view>
+#include <type_traits>
+#include <variant>
+
+using namespace std;
+
+#pragma warning(push)
+#pragma warning(disable : 4582) // 'uninit_vector::storage': constructor is not implicitly called
+#pragma warning(disable : 4583) // 'uninit_vector::storage': destructor is not implicitly called
+
+template <class CharType>
+constexpr auto get_input_literal() {
+    if constexpr (same_as<CharType, char>) {
+        return "meow";
+    } else {
+        return L"meow";
+    }
+}
+
+template <class CharType>
+constexpr auto get_input_sv() {
+    if constexpr (same_as<CharType, char>) {
+        return "meow"sv;
+    } else {
+        return L"meow"sv;
+    }
+}
+
+class context {
+public:
+    using char_type = char;
+};
+
+class wcontext {
+public:
+    using char_type = wchar_t;
+};
+
+enum class Arg_type : uint8_t {
+    none,
+    int_type,
+    unsigned_type,
+    long_type,
+    unsigned_long_type,
+    bool_type,
+    char_type,
+    float_type,
+    double_type,
+    long_double_type,
+    pointer_type,
+    string_literal_type,
+    string_type,
+    handle_type,
+};
+
+template <class Context>
+auto visitor = [](auto&& arg) {
+    using T         = decay_t<decltype(arg)>;
+    using char_type = typename Context::char_type;
+    if constexpr (is_same_v<T, monostate>) {
+        return Arg_type::none;
+    } else if constexpr (is_same_v<T, int>) {
+        return Arg_type::int_type;
+    } else if constexpr (is_same_v<T, unsigned>) {
+        return Arg_type::unsigned_type;
+    } else if constexpr (is_same_v<T, long long>) {
+        return Arg_type::long_type;
+    } else if constexpr (is_same_v<T, unsigned long long>) {
+        return Arg_type::unsigned_long_type;
+    } else if constexpr (is_same_v<T, char_type>) {
+        return Arg_type::char_type;
+    } else if constexpr (is_same_v<T, float>) {
+        return Arg_type::float_type;
+    } else if constexpr (is_same_v<T, double>) {
+        return Arg_type::double_type;
+    } else if constexpr (is_same_v<T, long double>) {
+        return Arg_type::long_double_type;
+    } else if constexpr (is_same_v<T, const void*>) {
+        return Arg_type::pointer_type;
+    } else if constexpr (is_same_v<T, const char_type*>) {
+        return Arg_type::string_literal_type;
+    } else if constexpr (is_same_v<T, basic_string_view<char_type>>) {
+        return Arg_type::string_type;
+    } else {
+        return Arg_type::handle_type;
+    }

```

Comment:
can use _Get_type once that's in, we need something that works like this in the product code anyways